### PR TITLE
Add Creaking configs

### PR DESCRIPTION
**Pull request by Onako2**
Checklist:
- [x] Add Rideables config
- [x] Add attributes such as scale

creaking_transient doesn't support riding. Even when I tried forcefully putting the player on a creaking_transient the player would desync. Creaking Transient will be removed anyway in 1.21.4.

### CODE REVIEW

File: patches/server/0007-Ridables.patch [hunk 1]
```diff
@@ -4750,6 +4750,70 @@ index 03e3cbe73119ca76417d4dd192e1560bdfc373ec..aa4160892b9ba2486806b5a54bc1465a
...
+         }
+diff --git a/src/main/java/net/minecraft/world/entity/monster/creaking/CreakingTransient.java b/src/main/java/net/minecraft/world/entity/monster/creaking/CreakingTransient.java
+index b57818cbe9ac6a2095da8a2ee6ccbd4c1804a502..17314fa27219c6a8871e7bd3962567f0627db886 100644
+--- a/src/main/java/net/minecraft/world/entity/monster/creaking/CreakingTransient.java
++++ b/src/main/java/net/minecraft/world/entity/monster/creaking/CreakingTransient.java
+@@ -38,6 +38,7 @@ public class CreakingTransient extends Creaking {
+         super(type, world);
+     }
+ 
++
```
Comment (granny) [inline patches/server/0007-Ridables.patch]: leftover whitespace

File: patches/server/0007-Ridables.patch [hunk 2]
```diff
@@ -5288,6 +5352,17 @@ index 42e502cfcb8d2e775cbf738773caf1a087d2f3f4..2e35e494134e5294fd67a0bf9abbe3f6
 +        cowControllable = getBoolean("mobs.cow.controllable", cowControllable);
 +    }
 +
++    public boolean creakingRidable = false;
++    public boolean creakingRidableInWater = true;
++    public boolean creakingControllable = true;
++    private void creakingSettings() {
++        if (net.minecraft.server.MinecraftServer.getServer().getWorldData().enabledFeatures().contains(net.minecraft.world.flag.FeatureFlags.WINTER_DROP)) { // TODO: Remove check when 1.21.4 Purpur releases
```
Comment (granny) [inline patches/server/0007-Ridables.patch]: Remove this if statement. There's no reason to hide config options from end-users.

**Review by granny — CHANGES_REQUESTED** [review]

**Review by granny — APPROVED** [review]
LGTM :) Will be merged once a PR is made to PurpurDocs with the new options.

**Comment by Onako2** [comment]
Ok, I planned anyway to do a PR to docs lol

**Pull request merged**